db/write_batch.cc

```diff
@@ -108,26 +110,20 @@ void WriteBatchInternal::SetSequence(WriteBatch* b, SequenceNumber seq) {
   EncodeFixed64(&b->rep_[0], seq);
 }
 
-void WriteBatch::Put(const Slice& key, const Slice& value) {
-  WriteBatchInternal::SetCount(this, WriteBatchInternal::Count(this) + 1);
-  rep_.push_back(static_cast<char>(kTypeValue));
-  PutLengthPrefixedSlice(&rep_, key);
-  PutLengthPrefixedSlice(&rep_, value);
-}
-
-void WriteBatch::PutWriteTime(const Slice& key, const Slice& value) {
-  WriteBatchInternal::SetCount(this, WriteBatchInternal::Count(this) + 1);
-  rep_.push_back(static_cast<char>(kTypeValueWriteTime));
-  PutLengthPrefixedSlice(&rep_, key);
-  PutVarint64(&rep_,Env::Default()->NowMicros());
-  PutLengthPrefixedSlice(&rep_, value);
-}
-
-void WriteBatch::PutExplicitExpiry(const Slice& key, const Slice& value, ExpiryTime expiry) {
+void WriteBatch::Put(const Slice& key, const Slice& value, const KeyMetaData * meta) {
+  KeyMetaData local_meta;
   WriteBatchInternal::SetCount(this, WriteBatchInternal::Count(this) + 1);
-  rep_.push_back(static_cast<char>(kTypeValueExplicitExpiry));
+  if (NULL!=meta)
+      local_meta=*meta;
+  rep_.push_back(static_cast<char>(local_meta.m_Type));
   PutLengthPrefixedSlice(&rep_, key);
-  PutVarint64(&rep_, expiry);
+  if (kTypeValueExplicitExpiry==local_meta.m_Type
```

Comment:
Might be a bit cleaner to add an `IsExpiryType()` method to the `KeyMetaData` struct and call it here (and possibly elsewhere) instead of doing explicit comparisons against `m_Type`.